### PR TITLE
feat: Precommit check for secrets leak and file linting 

### LINKED ISSUE #88: Precommit check for secrets leak
**Issue summary**

We have already had two incidents involving secret leaks (both Slack tokens and DO keys). Even though they were immediately revoked and no damage was done (except for the extra work needed to rotate them), we should prevent such class of issues in the future.

One approach is to have a pre-commit hook that checks the staged files for potential secrets. It's not bulletproof, there are likely ways for a creative engineer to circumvent that, but it should help mitigate potential disasters.

Acceptance criteria:
- [x] Committing a file containing a Slack/Digital Ocean token alerts the engineer that he's potentially leaking secrets and prevents it.

**Other information and links**

https://github.com/Yelp/detect-secrets/tree/master
https://github.com/Yelp/detect-secrets/blob/master/.pre-commit-hooks.yaml
https://pre-commit.com/

There might be a nicer way to do it, so a bit of research beforehand might be beneficial.

### PR DESCRIPTION
**Pull request by samuelarogbonlo**
**Summary of changes**
Changes introduced in this pull request:
- Pre-commit check for secrets leak and File linting
- added the docs for this feature. 

**Reference issue to close (if applicable)**

Closes #88 

**Other information and links**

### CODE REVIEW

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
Let's add the docs for this feature in this PR. I can't really review it without the bigger picture and usage.

**Review by LesnyRumcajs — COMMENTED** [review]
Dismissing my review so it's not blocked for the duration of my PTO.

**Review by lemmih — CHANGES_REQUESTED** [review]
The formatting changes make it difficult for me to see the genuine modifications. Mind moving the formatting to a different PR?

File: README.md [hunk 1]
```diff
@@ -20,6 +20,25 @@ This repository contains machine-readable specifications for the auxillilary ser
...
+
+To use the pre-commit hooks in this repository, follow these steps:
+
+- **Install Pip**: If you don't have Pip installed on your system, you can find installation instructions [here](https://pip.pypa.io/en/stable/installation/).
+
+
+- **Install the Pre-commit Hooks**: Run the following command in your project's directory to install the hooks:
+
+    ```bash
+    pre-commit install
```
Comment (lemmih) [inline README.md:37]: Is `pre-commit` something everyone has installed?

**Review by lemmih — CHANGES_REQUESTED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**